app/models/miq_server/environment_management.rb

```diff
@@ -26,12 +26,17 @@ def get_network_information
 
     def start_memcached
       # TODO: Need to periodically check the memcached instance to see if it's up, and bounce it and notify the UiWorkers to re-connect
-      return unless ::Settings.server.session_store.to_s == 'cache'
-      return unless MiqEnvironment::Command.supports_memcached?
-      require "#{Rails.root}/lib/miq_memcached" unless Object.const_defined?(:MiqMemcached)
+      return if !::Settings.server.session_store.to_s == 'cache' || !MiqEnvironment::Command.supports_memcached?
+
+      require Rails.root.join("lib/miq_memcached").to_s unless Object.const_defined?(:MiqMemcached)
+
       _svr, port = MiqMemcached.server_address.to_s.split(":")
       opts = ::Settings.session.memcache_server_opts.to_s
-      MiqMemcached::Control.restart!(:port => port, :options => opts)
+
+      if MiqMemcached::Config.new(opts).changed?(MiqMemcached::Control::CONF_FILE)
```

Comment:
`CONF_FILE` living in `::Control` rather than `::Config` feels a little odd, this might be cleaner if `MiqMemcached::Config` had a default conf file path